modules/services/mpdscribble.nix

```
@@ -0,0 +1,218 @@
+      Last.fm (formerly AudioScrobbler)
+    '';
+
+    proxy = lib.mkOption {
+      default = null;
+      type = lib.types.nullOr lib.types.str;
+      description = ''
+        HTTP proxy URL.
+      '';
+    };
+
+    verbose = lib.mkOption {
+      default = 1;
+      type = lib.types.int;
+      description = ''
+        Log level for the mpdscribble daemon.
+      '';
+    };
+
+    journalInterval = lib.mkOption {
+      default = 600;
+      example = 60;
+      type = lib.types.int;
+      description = ''
+        How often should mpdscribble save the journal file? [seconds]
+      '';
+    };
+
+    host = lib.mkOption {
+      default = (if mpdCfg.network.listenAddress != "any" then
+        mpdCfg.network.listenAddress
+      else
+        "localhost");
+      defaultText = lib.literalExpression ''
+        if config.${mpdOpt.network.listenAddress} != "any"
+        then config.${mpdOpt.network.listenAddress}
+        else "localhost"
+      '';
+      type = lib.types.str;
+      description = ''
+        Host for the mpdscribble daemon to search for a mpd daemon on.
+      '';
+    };
+
+    package = lib.mkPackageOption pkgs "mpdscribble" { };
+
+    passwordFile = lib.mkOption {
+      default = null;
+      type = lib.types.nullOr lib.types.str;
+      description = ''
+        File containing the password for the mpd daemon.
+      '';
+    };
+
+    port = lib.mkOption {
+      default = mpdCfg.network.port;
+      defaultText = lib.literalExpression "config.${mpdOpt.network.port}";
+      type = lib.types.port;
+      description = ''
+        Port for the mpdscribble daemon to search for a mpd daemon on.
+      '';
+    };
+
+    endpoints = lib.mkOption {
+      type = let
+        endpoint = { name, ... }: {
+          options = {
+            url = lib.mkOption {
+              type = lib.types.str;
+              default = endpointUrls.${name} or "";
+              description =
+                "The url endpoint where the scrobble API is listening.";
+            };
+            username = lib.mkOption {
+              type = lib.types.str;
+              description = ''
+                Username for the scrobble service.
+              '';
+            };
+            passwordFile = lib.mkOption {
+              type = lib.types.nullOr lib.types.str;
+              description =
+                "File containing the password, either as MD5SUM or cleartext.";
+            };
+          };
+        };
+      in lib.types.attrsOf (lib.types.submodule endpoint);
+      default = { };
+      example = {
+        "last.fm" = {
+          username = "foo";
+          passwordFile = "/run/secrets/lastfm_password";
+        };
+      };
+      description = ''
+        Endpoints to scrobble to.
+        If the endpoint is one of "${
+          lib.concatStringsSep ''", "'' (builtins.attrNames endpointUrls)
+        }" the url is set automatically.
+      '';
+    };
+
+  };
+
+  config = lib.mkIf cfg.enable {
```

```suggestion
  config = lib.mkIf cfg.enable {
      assertions = [
      (lib.hm.assertions.assertPlatform "services.mpdscribble" pkgs
        lib.platforms.linux)
    ];
```
sorry, came back and realized this was a linux only module.